lib/util/schema-utils.ts

```diff
@@ -1,94 +1,127 @@
 import { z } from 'zod';
 
-export function looseArray<T extends z.ZodTypeAny>(
-  schema: T,
-  catchCallback?: () => void
-): z.ZodEffects<
-  z.ZodCatch<z.ZodArray<z.ZodCatch<z.ZodNullable<T>>, 'many'>>,
-  z.TypeOf<T>[],
-  unknown
-> {
-  type Elem = z.infer<T>;
+interface ErrorContext<T> {
+  error: z.ZodError;
+  input: T;
+}
 
-  const nullableSchema = schema.nullable().catch(
-    catchCallback
-      ? () => {
-          catchCallback();
-          return null;
+interface LooseOpts<T> {
+  onError?: (err: ErrorContext<T>) => void;
+}
+
+/**
+ * Works like `z.array()`, but drops wrong elements instead of invalidating the whole array.
+ *
+ * **Important**: non-array inputs are still invalid.
+ * Use `LooseArray(...).catch([])` to handle it.
+ *
+ * @param Elem Schema for array elements
+ * @param onError Callback for errors
+ * @returns Schema for array
+ */
+export function LooseArray<Schema extends z.ZodTypeAny>(
```

Comment:
why uppercase function name?

Reply:
I think it communicates better the intent of being like a generic, but for schema

Reply:
🤔 @rarkins @JamieMagee @secustor WDYT?

Reply:
I personally would expect lowercase, but IMO this comes down to personal taste.